### ISSUE TITLE
Regent: index launch with disjoint elements of partition

### ISSUE THREAD

**Issue by rauteric** [StanfordLegion/legion#607]
Consider the following:
```
import "regent"
  
task test(x: region(ispace(int1d),double), y: region(ispace(int1d),double))
where
  reads(x), writes(y)
do
  -- ...
end

task main()
  var reg = region(ispace(int1d,4),double)
  var part = partition(equal,reg,ispace(int1d,4))

  fill(reg,0.0)

  __demand(__index_launch)
  for i = 1,4 do
    test(part[0],part[i])
  end
end

regentlib.start(main)
```

The equal partitions should be disjoint, but with `__demand(__index_launch)`, Regent complains that the partitions interfere. If I change `writes(y)` to `reads(y)` in the test task, the index launch is allowed.

Is something like this possible in Regent? In particular, I'm interested in modifying the Cholesky example to use index task launches, as recommended in [this comment](https://github.com/StanfordLegion/legion/issues/574#issuecomment-524483902), where I think one would run into the issue above.

**Comment by elliottslaughter** [comment]
Talking with @magnatelee I think the most expedient way to do this will be to add a flag like `-foverride-demand-index-launch 1` to make the compiler trust launches like this one. At the moment, this is problematic because the compiler isn't smart enough to realize that `i > 0` (based on the loop bounds). That is the longer term answer, but I think we don't necessarily want to wait for that all to land. The override flag should be relatively quick to implement.

**Comment by elliottslaughter** [comment]
This flag is now available in master as of commit a5c903d448fa35abdd59ae4efd9e5e0f85615cac. This should be sufficient to rewrite Cholesky as suggested in the other thread.
